pkg/generate/app/componentresolvers.go

```diff
@@ -75,6 +75,15 @@ func (r PerfectMatchWeightedResolver) Resolve(value string) (*ComponentMatch, er
 	}
 	switch len(imperfect) {
 	case 0:
+		if isFile(value) {
```

Comment:
This file looks quite complex to understand, @jprukner could you add a comment here explaining why we do this check?

```go
case 0:
    // If value is a file and there is a TemplateFileSearcher in one of the resolvers
    // and trying to use it gives an error, use this error instead of ErrNoMatch.
    // E.g., calling `oc new-app template.json` where template.json is a file
    // with invalid JSON, it's better to return the JSON syntax error than a more
    // generic message.
    if isFile(value) {
```